.github/workflows/build-mysql-container.yaml

```diff
@@ -0,0 +1,136 @@
+      - name: Check TAG file
+        working-directory: containers
+        run: |
+          result="$(./tag_exists moco/mysql mysql/${{ matrix.mysql-version }})"
+          if [ "$result" = ok ]; then
+            exit 1
+          fi
+          echo "TAG=$(cat ./mysql/${{ matrix.mysql-version }}/TAG)" >> $GITHUB_ENV
+
+      - uses: docker/build-push-action@v4
+        with:
+          context: containers/mysql/${{ matrix.mysql-version }}/.
+          push: false
+          load: true
+          tags: |
+            ghcr.io/cybozu-go/moco/mysql:${{ env.TAG }}
+            ghcr.io/cybozu-go/moco/mysql:${{ matrix.mysql-version }}
+
+      - name: Install Container Structure Tests
+        run: |
+          VERSION=v1.15.0
+          curl -LO "https://storage.googleapis.com/container-structure-test/$VERSION/container-structure-test-linux-amd64" \
+            && chmod +x container-structure-test-linux-amd64 \
+            && sudo mv container-structure-test-linux-amd64 /usr/local/bin/container-structure-test
+      - name: Run Container Structure Tests
+        run: |
+          container-structure-test test --image ghcr.io/cybozu-go/moco/mysql:${{ matrix.mysql-version }} --config ./containers/mysql/${{ matrix.mysql-version }}/container-structure-test.yaml
+
+      - name: Create kind cluster
+        run: kind create cluster --name=moco --config=./e2e/kind-config_actions.yaml --image=kindest/node:v${{ matrix.k8s-version }} --wait 1m
+
+      - name: Load mysqld image
+        run: kind load docker-image ghcr.io/cybozu-go/moco/mysql:${{ matrix.mysql-version }} --name moco
+
+      - name: Install MOCO
+        run: |
+          kubectl apply -f https://github.com/jetstack/cert-manager/releases/latest/download/cert-manager.yaml
+          kubectl -n cert-manager wait --for=condition=available --timeout=180s --all deployments
+          kubectl apply -f https://github.com/cybozu-go/moco/releases/latest/download/moco.yaml
+          kubectl -n moco-system wait --for=condition=available --timeout=180s --all deployments
+
+      - name: Create MySQLCluster
+        run: |
+          cat > mycluster.yaml <<'EOF'
+          apiVersion: moco.cybozu.com/v1beta1
+          kind: MySQLCluster
+          metadata:
+            namespace: default
+            name: test
+          spec:
+            replicas: 3
+            podTemplate:
+              spec:
+                containers:
+                  - name: mysqld
+                    image: ghcr.io/cybozu-go/moco/mysql:${{ matrix.mysql-version }}
+            volumeClaimTemplates:
+              - metadata:
+                  name: mysql-data
+                spec:
+                  accessModes: [ "ReadWriteOnce" ]
+                  resources:
+                    requests:
+                      storage: 1Gi
+          EOF
+          kubectl apply -f mycluster.yaml
+
+      - name: Wait for MySQLCluster
+        run: kubectl wait -n default --for=condition=Available mysqlcluster/test --timeout=180s
```

Comment:
Currently we are only testing until MySQLCluster is ready.
When the MySQL image is pushed to ghcr.io, we will rewrite the E2E manifests and modify it to run the E2E test using the built mysqld image.

https://github.com/cybozu-go/moco/blob/main/e2e/testdata/single.yaml#L26